Ecal/src/Ecal/EcalVetoProcessor.cxx

```diff
@@ -1042,13 +1044,21 @@ void EcalVetoProcessor::produce(framework::Event &event) {
           }
         }
 
-        // Perform "linreg" on selected points:
-        TDecompSVD svdObj = TDecompSVD(hdt);
+        // Perform "linreg" on selected points
+        // Calculate the determinant of the matrix
+        double determinant =
+            hdt(0, 0) * (hdt(1, 1) * hdt(2, 2) - hdt(1, 2) * hdt(2, 1)) -
+            hdt(0, 1) * (hdt(1, 0) * hdt(2, 2) - hdt(1, 2) * hdt(2, 0)) +
+            hdt(0, 2) * (hdt(1, 0) * hdt(2, 1) - hdt(1, 1) * hdt(2, 0));
```

Comment:
Haha I know! It's ridiculous... Calling this `.Determinant();` function will create a similar error msg that I wanted to avoid in the first place, i.e. ROOT internal and something we cannot silence. There is also a way to check the `.Condition()` of the matrix, which could serve a similar purpose, you wont guess... that will also lead to a ROOT internal msg...

Reply:
Well, we _could_ switch to Eigen which is already installed in the development image for acts. It has an [SVD decomposition](https://eigen.tuxfamily.org/dox/group__LeastSquares.html), [a determinant member function](https://eigen.tuxfamily.org/dox/classEigen_1_1MatrixBase.html#a7ad8f77004bb956b603bb43fd2e3c061), and compile-time dimensions which might help improve performance.

Reply:
Sure, but I'd do that in another PR tho.